knotx-core/knotx-repository/src/main/java/com/cognifide/knotx/repository/impl/HttpRepository.java

```diff
@@ -73,19 +70,20 @@ static HttpRepository of(String path, String domain, Integer port, Vertx vertx)
                 );
     }
 
-    private Observable<Buffer> reduceBuffers(HttpClientResponse response) {
+    private Observable<RepositoryResponse> processResponse(final HttpClientResponse response) {
         return Observable.just(Buffer.buffer())
                 .mergeWith(response.toObservable())
-                .reduce(Buffer::appendBuffer);
+                .reduce(Buffer::appendBuffer)
+                .flatMap(buffer -> toRepositoryResponse(buffer,response));
     }
 
-    private Observable<RepositoryResponse> toRepositoryResponse(Buffer buffer) {
+    private Observable<RepositoryResponse> toRepositoryResponse(Buffer buffer, final HttpClientResponse httpClientResponse) {
         Observable<RepositoryResponse> response;
         if (buffer.length() > 0) {
             response = RepositoryResponse.success(buffer.toString()).toObservable();
```

Comment:
where it is fixed ? I don't see RepositoryResponse.success() consuming response headers here